api/v1/endpoints/listings.py

```diff
@@ -26,44 +29,38 @@ def get(self):
         """
         Return a list of listings
         """
-        args = endpoint_arguments.parse_args(request)
-        page = args.get('page', 1)
-        per_page = args.get('per_page', 10)
-
-        listings = ''
-        return listings
+        # args = endpoint_arguments.parse_args(request)
+        # page = args.get('page', 1)
+        # per_page = args.get('per_page', 10)
+        db = dynamo.dynamo_conn
+        listings = db.get_all_listings()
+        print(listings)
+        return {'items': listings}, 200
 
 @ns.route('/', methods=['POST'])
 class Listing(Resource):
     @api.expect(listing_arguments)
-    @api.response(201, 'Listing successfully added')
-    @api.response(500, 'Listing failed due to server side error')
+    @api.marshal_with(new_listing)
+    @api.response(201, constants.NEW_LISTING_SUCCESS)
+    @api.response(400, constants.MISSING_PAYLOAD_DATA)
+    @api.response(500, constants.SERVER_ERROR)
     def post(self):
         """
         Persist a new listing to file storage, db, and protocol
         """
         timings = {}
         start_time = time.time()
-        file_type = None
-        md5_sum = None
-        tags = None
+        payload = {}
         uploaded_md5 = None
-        filenames = []
-        listing_hash = None
-        if not request.form.get('listing_hash'):
-            return 'No listing hash provided', 400
-        else:
-            listing_hash = request.form.get('listing_hash')
-        if not request.form.get('file_type'):
-            return 'No file type specified', 400
-        else:
-            file_type = request.form.get('file_type')
-        if not request.form.get('md5_sum'):
-            return 'No md5 sum provided', 400
-        else:
-            md5_sum = request.form.get('md5_sum')
+        for item in ['title', 'description', 'license', 'file_type', 'md5_sum', 'listing_hash']:
+            if not request.form.get(item):
+                api.abort(400, (constants.MISSING_PAYLOAD_DATA % item))
+            else:
+                payload[item] = request.form.get(item)
         if request.form.get('tags'):
-            tags = request.form.get('tags')
+            payload['tags'] = [x.strip() for x in request.form.get('tags').split(',')]
+        filenames = []
+        md5_sum = request.form.get('md5_sum')
         if request.form.get('filenames'):
             filenames = request.form.get('filenames').split(',')
```

Comment:
is there a guarantee that any filenames['foo', 'bar'] match files [bar.whatever, foo.whatever] ?

Reply:
It's an exact match only, so only if `bar` is the submitted filename, it will only match the exact text `bar`. (if I'm following your question correctly)

Reply:
question was due to the fact that `files` and `filenames` are submitted separately right? My concern was that somethere there were files [file_one, file_two] and somewhere else there were names [bar, foo] and that they didn't match up. 

maybe its a non concern?

Reply:
Ah, I gotcha. Currently, everything is done at the same time. The file is submitted along with the metadata and file name